### PR TITLE
[query] fix AzureStorageFS openNoCompression 

### PR DESCRIPTION
**Pull request by vrautela**
- fix double-counting error in openNoCompression method 
- add tests to read/write more bytes than can be held in the ByteBuffer
- abstract out seek method into FS

### CODE REVIEW

File: hail/src/main/scala/is/hail/io/fs/FS.scala [hunk 1]
```diff
@@ -151,6 +151,15 @@ abstract class FSSeekableInputStream extends InputStream with Seekable {
     toTransfer
   }
 
+  def seek(newPos: Long): Unit = {
+    bb.clear()
+    bb.limit(0)
+    pos = newPos
+    adjustSeekInternalState(newPos)
+  }
```
Comment (daniel-goldstein) [inline hail/src/main/scala/is/hail/io/fs/FS.scala:159]: So I think I've slightly changed my mind here about this extra method approach. I think we should just keep seek like it is here without the call to `adjustSeekInternalState`, and then the `GoogleStorageFS` should override `seek` with a call to `super.seek` and whatever additional bookkeeping we're currently putting in `adjustSeekInternalState`.

File: hail/src/test/scala/is/hail/fs/FSSuite.scala [hunk 2]
```diff
@@ -271,6 +271,60 @@ trait FSSuite {
...
+
+    fs.delete(f, false)
+
+    assert(!fs.exists(f))
+  }
+
+  @Test def testReadWriteBytesLargerThanBuffer(): Unit = {
+    val f = t()
+
+    val byteBufferSize = 64 * 1024
```
Comment (daniel-goldstein) [inline hail/src/test/scala/is/hail/fs/FSSuite.scala:295]: This test is a bit brittle because it depends on an exact value of an implementation detail of the FS. If we were to change the implementation, the test could become invalid, and we might forget to update the tests. I think it is unlikely that we would ever use a buffer size greater than, say ten megabytes, so let's instead ensure that our tests cover a spread of different sizes that are likely to catch this edge case regardless of the exact buffer size.

**Review by daniel-goldstein — CHANGES_REQUESTED** [review]

File: hail/src/test/scala/is/hail/fs/FSSuite.scala [hunk 3]
```diff
@@ -271,6 +271,60 @@ trait FSSuite {
...
+    assert(fs.exists(f))
+
+    using(fs.open(f)) { is =>
+      assert(is.read() == 1)
+      assert(is.read() == 127)
+      assert(is.read() == 255)
+
+      var i = 0
+      while (i < byteBufferSize + 10) {
+        assert(is.read() == i)
```
Comment (daniel-goldstein) [inline hail/src/test/scala/is/hail/fs/FSSuite.scala:318]: This value will quickly exceed the maximum value for a byte. What would you expect to read back in such a case?

**Comment by daniel-goldstein** [comment]
Merging this inside #11816
